src/session_state.rs

```diff
@@ -11,25 +13,44 @@ use embedded_time::{
 /// The default duration to wait for a ping response from the broker.
 const PING_TIMEOUT: Seconds = Seconds(5);
 
-pub struct SessionState<Clock: embedded_time::Clock, const MSG_SIZE: usize, const MSG_COUNT: usize>
-{
+pub struct MessageRecord<const N: usize> {
+    pub msg: Vec<u8, N>,
+    transmitted: bool,
+    _id: u16,
+    qos: QoS,
+}
```

Comment:
```suggestion
pub struct MessageRecord<const N: usize> {
    transmitted: bool,
    _id: u16,
    qos: QoS,
    pub msg: Vec<u8, N>,
}
```

To help the compiler with the `MaybeUninit` in `Vec` a bit.

Also what does `_id` do?